velox/connectors/hive/storage_adapters/s3fs/tests/S3FileSystemTest.cpp

```diff
@@ -288,4 +299,35 @@ TEST_F(S3FileSystemTest, invalidConnectionSettings) {
   VELOX_ASSERT_THROW(
       filesystems::S3FileSystem("", hiveConfig), "Invalid duration");
 }
+
+TEST_F(S3FileSystemTest, registerCredentialProviderFactories) {
+  const auto credentialProvider = "my-credential-provider";
+  const auto invalidCredentialProvider = "invalid-credential-provider";
+  registerAWSCredentialsProvider(
+      credentialProvider, [](const S3Config& config) {
+        return std::make_shared<MyCredentialsProvider>();
+      });
+
+  auto hiveConfig = minioServer_->hiveConfig(
+      {{"hive.s3.credentials-provider", credentialProvider}});
+  ASSERT_NO_THROW(filesystems::S3FileSystem("", hiveConfig));
+
+  // Configure with unregistered credential provider.
+  hiveConfig = minioServer_->hiveConfig(
+      {{"hive.s3.credentials-provider", invalidCredentialProvider}});
+  VELOX_ASSERT_THROW(
+      filesystems::S3FileSystem({"", hiveConfig}),
+      fmt::format(
+          "CredentialsProviderFactory for '{}' not registered",
```

Comment:
We can directly use `CredentialsProviderFactory for 'invalid-credential-provider' not registered`